packages/headless/src/app/engine-configuration-options.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  BooleanValue,
+  RecordValue,
+  SchemaDefinition,
+  StringValue,
+} from '@coveo/bueno';
+import {
+  AnalyticsClientSendEventHook,
+  IRuntimeEnvironment,
+} from 'coveo.analytics';
+import {PreprocessRequest} from '../api/preprocess-request';
+
+/**
+ * The global headless engine configuration options.
+ */
+export interface EngineConfigurationOptions {
+  /**
+   * The unique identifier of the target Coveo Cloud organization (e.g., `mycoveocloudorganizationg8tp8wu3`)
+   */
+  organizationId: string;
+  /**
+   * The access token to use to authenticate requests against the Coveo Cloud endpoints. Typically, this will be an API key or search token that grants the privileges to execute queries and push usage analytics data in the target Coveo Cloud organization.
+   */
+  accessToken: string;
+  /**
+   * A function that fetches a new access token. The function must return a Promise that resolves to a string (the new access token).
+   */
+  renewAccessToken?: () => Promise<string>;
```

Comment:
I know token renewal is supported for search tokens, but I can't remember whether it is supported by Coveo Cloud at large. Do you know?

Reply:
Based on my understanding, renewing a token is not supported by the Coveo platform (e.g. using refresh tokens). An implementer needs to create a function that calls [this endpoint](https://platform.cloud.coveo.com/docs?urls.primaryName=Search%20API#/Search%20V2/token) using their API key.